### ISSUE TITLE
Support default settings for mute/unmute and camera off/on

### ISSUE THREAD

**Issue by philipreimer** [nextcloud/spreed#3270]
**Is your feature request related to a problem? Please describe.**
Everytime I join a Nextcloud Talk call using the browser, microphone and camera are active by default. I always have to mute myself and disable the camera.
I have made the experience that joining a call muted is favorable, especially in group calls, and especially if not everyone uses a headset. When everyone joins unmuted there often is a lot of disturbing background noise.

**Describe the solution you'd like**
There should be a way to configure the default settings for microphone and camera, which are applied automatically when joining a call.

A new section in the Personal Settings dedicated to Talk could be created containing two checkboxes:
* join calls muted by default
* join calls with camera off by default

**Describe alternatives you've considered**

The global Talk settings, which can only be changed by Administrators, could be extended to include two checkboxes:
* call participants join muted by default
* call participanst join with camera off by default

**Comment by wbob** [comment]
I have the opposite experience for one-on-one chats: when joining calls after allowing mic+cam in the browser, I have to activate the inputs by clicking the symbols each, something I need to explain to a user unfamiliar with the interface initially. In one-on-ones, it's clear we both want to talk right away.

Though I understand your argument for a different default for group calls. I somehow think browsers should have a default handling, but couldn't find if the behaviour is meant to be standardized.

**Comment by nickvergessen** [comment]
@wbob just upgrade your Talk to 8.0.7, that comes with default enabled and remembers the state in your browser on a per conversation level.

**Comment by wbob** [comment]
@nickvergessen - very kind to hook me up with #3205, thank you. I really tried to search for prior issues but didn't land on #2485.

On topic: one proposition to the aforementioned issue of @philipreimer could be to make use of more than one join-button ("join with audio&camera", "join with audio only", "join muted") for group calls as surfacing as "topbar" discussion also in #2969. It's an interesting UX challenge.

**Comment by philipreimer** [comment]
@wbob @nickvergessen I missed #2485 when looking for existing issues. Reading the comments there it seems that people think having a setting to start calls muted by default might be confusing for the user. So the idea is now to have separate buttons when joining a call? Is there a specific issue that I could follow?

Then this could be closed.

**Comment by nickvergessen** [comment]
No, I guess we will use this issue for that topic.

**Comment by kzar79** [comment]
It seems to me that different people need different defaults. In my case, I need to automatically activate audio AND video for all kind of calls, because my users are confused by the fact that they talk and I cannot see nor hear them, and it is difficult to explain they need to click a couple of buttons. They don't even find the buttons...

What I propose is to add two option for the administrator (or two options in each call, even better) for what to use as a default when starting/joining a call:
- "Audio default enabled/disabled"
- "Video default enabled/disabled"

I don't think it would be difficult. If needed I can work on it myself (but please first accept my proposed solution so I don't create a merge for nothing).

**Comment by Debianer** [comment]
I just tested with about 20 people trying to join. Most of them were thrown out and the whole system never got to work. I believe this a performance issue, since the server is running with the high performance backend and there was little load on the turn server and the high performance backend.

If people could join without their camera and microphone switched on, a lot more people would be able to join a call and listen to a few speakers.

**Comment by PVince81** [comment]
> No, I guess we will use this issue for that topic.

@nickvergessen so split the button to be "Join with audio only" and "Join with video" like on mobile ?
There might be some design challenges to make it look nice and compact (cc @ma12-co)

I guess 20 people joining with audio only (unmuted) should have less performance impact.

Personally I usually prefer to start muted and no video and would vote for having at least a personal option as default for all calls.

**Comment by nickvergessen** [comment]
There are discussions on going and it will be topic in the Design call for the Talk app next week

**Comment by jancborchardt** [comment]
There’s 3 areas to this:
- Microphone mute/unmute and video on/off should be remembered per conversation
- We need a pre-join interstitial screen where you see your camera and feedback about microphone working (or not) and where you can enable/disable the devices. Jitsi e.g. does this and it’s very clear feedback.
- What we can additionally think about is a setting to "Always join muted" and "Always join with video off". If we do the previous points well, this might not be needed.

**Comment by schiessle** [comment]
I think in most cases the default regarding the mic are OK. If I join a call I usually want to talk. With video it is already more tricks, I saw people multiple times surprised that everyone can see them directly after they join.

On the other hand, in privacy sensitive industries a "everything off" by default can make definitely sense. I think a admin setting for the defaults with the option to overwrite it on a user level, similar to how we handle the activity settings, could make sense.

Just some feedback from my side, from what I heard by talking to users.

**Comment by PVince81** [comment]
> Microphone mute/unmute and video on/off should be remembered per conversation (thought this is the case already?)

Yes, that's already the case and is working. The state is stored in the browser's localStorage though, so if you switch browsers it's not there.

**Comment by philipreimer** [comment]
> The state is stored in the browser's localStorage though, so if you switch browsers it's not there.

Or if you clear your browser data.

**Comment by nickvergessen** [comment]
Or joined in a private tab

**Comment by nimisha-vijay** [comment]
How about a modal like this to check your audio/video and set devices? A similar design can be used in the settings as well. 

![image](https://user-images.githubusercontent.com/52440189/128481176-42885bb3-a83a-4a2a-9e43-2884048529be.png)

- [ ] When you click on "join call" (or "start call") in the top-right of a channel, this modal can appear where you check your camera and mic
- [ ] The camera/mic settings can be remembered for a particular channel
- [ ] There can be a little microphone meter that shows the sound level from your mic if it's on (as it's currently there for your own video in a Talk video call)
- [ ] On clicking "choose devices" you can change the camera or mic through a dropdown (If the name of the device is too long then it can wrap to the next line, the width of the dropdown should be limited to the width of the camera viewport)
- [ ] The "join call" button text can be different depending on what is on: only camera, only mic, both, or neither
- [ ] If you're starting a call in a channel instead of joining an existing one then the button text can be changed to "start call"

![image](https://user-images.githubusercontent.com/52440189/128481310-a11e8494-4fd7-41c9-b78f-7eca477c48d6.png)

What do you think? also cc @jancborchardt for the designs and any other feedback

**Comment by marcoambrosini** [comment]
I like it, though I still think we can and should find a way to let the user skip this step
